lib/lib-audit/src/main/java/org/sagebionetworks/audit/dao/ObjectRecordDAOImpl.java

```diff
@@ -0,0 +1,101 @@
+package org.sagebionetworks.audit.dao;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.sagebionetworks.aws.utils.s3.KeyGeneratorUtil;
+import org.sagebionetworks.aws.utils.s3.BucketDao;
+import org.sagebionetworks.aws.utils.s3.BucketDaoImpl;
+import org.sagebionetworks.aws.utils.s3.GzipCsvS3ObjectReader;
+import org.sagebionetworks.aws.utils.s3.GzipCsvS3ObjectWriter;
+import org.sagebionetworks.repo.model.audit.ObjectRecord;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.amazonaws.services.s3.AmazonS3Client;
+
+public class ObjectRecordDAOImpl implements ObjectRecordDAO {
+
+	private final static String[] HEADERS = new String[]{"timestamp", "changeNumber", "objectType", "jsonString"};
+
+	@Autowired
+	private AmazonS3Client s3Client;
+	
+	/**
+	 * Injected via Spring
+	 */
+	private int stackInstanceNumber;
+	/**
+	 * Injected via Spring
+	 */
+	private String objectRecordBucketFormat;
+	
+	private GzipCsvS3ObjectReader<ObjectRecord> reader;
+	private GzipCsvS3ObjectWriter<ObjectRecord> writer;
+	private Map<String, BucketDao> bucketDaoMap;
+	private Map<String, String> bucketNameMap;
+	
```

Comment:
Create an object that keeps track of the map and initiates bucketDao when needed. Take out type check.